README.md

```diff
@@ -1,16 +1,110 @@
-## exclusive-lock-node
+<!-- TOC -->
 
+- [`ExclusiveLock`](#exclusivelock)
+- [Installation](#installation)
+- [Usage](#usage)
+- [API](#api)
+  - [`new ExclusiveLock(options)`](#new-exclusivelockoptions)
+  - [`acquire()`](#acquire)
+  - [`inspect()`](#inspect)
+  - [`release()`](#release)
+  - [`refresh()`](#refresh)
+- [Authors](#authors)
+- [License](#license)
 
+<!-- /TOC -->
+## `ExclusiveLock`
+
+Need to synchronize a bunch of workers so that only 1 does a particular thing?  If yes,
+then you need a lock file.  This package uses a cache store (`redis` or `keydb`)
+to create records atomically that only 1 instance will "own."
 ## Installation
 
+```bash
+npm install @logdna/exclusive-lock --save
+```
 
 ## Usage
 
+```js
+const ExclusiveLock = require('@logdna/exclusive-lock')
+const cache_connection = require('./my-keydb-connection.js')
+const pino = require('pino')
+
+const exclusive_lock = new ExclusiveLock({
+  app_name: 'my-distributed-app'
+, log: pino()
+, cache_manager
+})
+
+async function main() {
+  const got_lock = await exclusive_lock.acquire()
+  if (got_lock) {
+    log.info('You win the race! Lock acquired')
```

Comment:
Is it possible to check whether the refresh failed?

That way we can make this case more rare:
- lock acquired
- long running sync task
- record expire, other instance get the lock
- the original task continues to run.

For example

```javascript
const lock = await exclusive_lock.acquire()
if (lock.acquired) {
  try {
    while (lock.acquired) {
      // do something that takes a lot of time
    }
  } finally {
    lock.release() // internally, only expire the record if it matches my version
  }
}
```

Reply:
The refresh is a bit tricky because in most cases it's run via a timer.  In the case where someone says `auto_refresh: false`, then it'll be up to them to call `refresh` when desired, and for that we could return the promise.  This is a valuable change.

I think the best of both worlds will be to emit an event on failure if it's happening asynchronously.  The logging is ok, but not programmatically sufficient to know about errors.